spring-kafka/src/main/java/org/springframework/kafka/core/KafkaOperations.java

```diff
@@ -64,6 +64,16 @@
 	ListenableFuture<SendResult<K, V>> sendDefault(int partition, K key, V data);
 
 	/**
+	 * Send the data to the default topic with the provided key and partition.
+	 * @param partition the partition.
+	 * @param timestamp the timestamp of the record.
+	 * @param key the key.
+	 * @param data the data.
+	 * @return a Future for the {@link SendResult}.
+	 */
+	ListenableFuture<SendResult<K, V>> sendDefault(int partition, Long timestamp, K key, V data);
```

Comment:
Ha! really, `ConsumerRecord` uses primitives for `timestamp` as well as for `partition`.
Well, I think I know why it is like that.
You can send a record with `null` `partition` and Kafka will assign one by the `key` (if present) or using round-robin algorithm otherwise.
Something similar happens for the timestamp if you send `null` - the broker assigns one using its local time.

My concern here was about inconsistency of our primitive type for the `partition` and how `ProducerRecord` expects wrapper.

So, my proposition is to change it into `Integer` as well to be consistent with Kafka API. 
Anyway this `timestamp` addition provides breaking change.